scripts/preview-theme.js

```diff
@@ -30,160 +99,365 @@ function findCommentPredicate(inputs, comment) {
-}
-
-const themeContribGuidelines = `
-  \rHi, thanks for the theme contribution, please read our theme [contribution guidelines](https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md#themes-contribution).
-  \rWe are currently only accepting color combinations from any VSCode theme or themes which have good color combination to minimize bloating the themes collection.
-
-  \r> Also note that if this theme is exclusively for your personal use, then instead of adding it to our theme collection you can use card [customization options](https://github.com/anuraghazra/github-readme-stats#customization)
-`;
-
-async function run() {
+};
+
+/**
+ * Retrieve javascript object from json string.
+ *
+ * @description Wraps the Hjson parse function to fix several known json syntax errors.
+ *
+ * @param {string} json The json to parse.
+ * @returns {Object} Object parsed from the json.
+ */
+const parseJSON = (json) => {
   try {
-    const ccc = new ColorContrastChecker();
-    const warnings = [];
-    const token = core.getInput("token");
-    const octokit = github.getOctokit(token || process.env.PERSONAL_TOKEN);
-    const pullRequestId = getPrNumber();
-
-    if (!pullRequestId) {
-      console.log("PR not found");
-      return;
+    return Hjson.parse(json);
+  } catch (error) {
+    const parsedJson = json
+      .split(/(},?)(?=\s*["|'][a-z]+["|'])/)
+      .filter((x) => typeof x !== "string" || !!x.trim());
+    if (parsedJson[0] === "},") {
+      parsedJson.push(parsedJson.shift());
+      return Hjson.parse(parsedJson.join(""));
+    } else {
+      throw error;
     }
+  }
+};
+
+/**
+ * Convert array to human readable comma separated string.
+ *
+ * @param {string[]} arr Input array.
+ * @param {string} [wrapSymbol=""] Wrap symbol.
+ * @returns {string} Human readable comma separated string.
+ */
+export const array2str = (arr, wrapSymbol = "") => {
+  if (arr.length === 0) {
+    return "";
+  } else if (arr.length === 1) {
+    return wrapSymbol ? "'" + arr[0] + "'" : arr[0];
+  }
+  if (wrapSymbol) arr = arr.map((x) => "'" + x + "'");
+  return `${arr.slice(0, arr.length - 1).join(", ")} and ${
+    arr[arr.length - 1]
+  }`;
+};
+
+/**
+ * Check if string is a valid hex color.
+ *
+ * @param {string} str String to check.
+ * @returns {boolean} Whether the string is a valid hex color.
+ */
+const isHexColor = (str, prefix = false) => {
+  if (prefix) {
+    return /^#[a-f0-9]{6}$/i.exec(str);
+  } else {
+    return /^[a-f0-9]{6}$/i.exec(str);
+  }
+};
+
+/**
+ * Check whether the theme name is still available.
+ * @param {string} name Theme name.
+ * @returns {boolean} Whether the theme name is available.
+ */
+const themeNameAlreadyExists = (name) => {
+  return themes[name] !== undefined;
+};
+
+/**
+ * Main function.
+ * @param {string} pull_request The pull request number to check. If empty action context or environment variable will be used.
+ */
+export const run = async (pull_request) => {
```

Comment:
The `pull_request` argument is added here so that we can use this script to check all theme PR at once.